### DISCUSSION TITLE
[papaparse] js-doc commit breaks complete() on config parsing

### DISCUSSION THREAD

**Discussion by sean-daley** [DefinitelyTyped/DefinitelyTyped#56679] [Issues with a @types package] [answered]
Starting with version 5.3.0 I now get typescript errors around our usage of complete()
```
src/containers/Jobs/JobRunDetails/annotations/EditAnnotations.spec.tsx:399:16 - error TS2554: Expected 2 arguments, but got 1.

399         config.complete(fileUploadResult)
                   ~~~~~~~~~~~~~~~~~~~~~~~~~~

  node_modules/@types/papaparse/index.d.ts:255:40
    255     complete?(results: ParseResult<T>, file: TInput): void;
                                               ~~~~~~~~~~~~
    An argument for 'file' was not provided.
```

It looks like as part of https://github.com/DefinitelyTyped/DefinitelyTyped/pull/56479 we lost the optional on the second file argument.

Old Definition:
```
complete?(results: ParseResult<T>, file?: File): void; // default: undefined
```

New Definition:
```
complete?(results: ParseResult<T>, file: TInput): void;
```

[a comment by an automated account is not shown]

**Comment by peterblazejewicz** [comment]
@sean-daley I understand the intent was to narrow type to case when input file is parsed (so the second param is present):
e.g: https://github.com/mholt/PapaParse/blob/master/papaparse.js#L556
Can you post code snippet you're using? thx!

**Comment by peterblazejewicz** [comment]
Please see this one: #56659

**Comment by sean-daley** [comment]
Unfortunately I cannot write everything but in our case what we are pretty much doing is the following:

1. We are passing in a custom `onComplete` method which looks like this:
```
Papa.parse(csvFile, { complete: onComplete, delimiter: ',' })
...
  const onComplete = (csvResult: ParseResult<string[]>): void => {
...
```

And then in some of our tests we are trying to mock the parse out at one point by doing:
```
    jest.spyOn(Papa, 'parse').mockImplementation((file, config): any => {
      if (config && config.complete) {
        config.complete(fileUploadResult)
      }
    })
```

Since we're only seeing these errors in our mocks (since we don't call complete directly in our code) I can probably just pass in `undefined` as a second argument and be done with it. But this is a change in behavior from before because before it was optional and now you have to at least specify undefined for the second argument. Though we can write our own custom onComplete method without adding in a second argument which seems a little weird.

Our bigger issue was the one you just fixed in https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/56659. After grabbing that in 5.3.1 I noticed this issue so if you think this should be correct behavior now, I can just update our test code to handle it.

**Comment by OpportunityLiu** [comment, marked as answer]
The `file` part is not optional in file parsing. I'm just updating the types to map the actual behavior of the lib.
